File: go/api/Language.go
package api

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// Language 对应 Language.py。
type Language struct {
	*session.ApiGroup
}

// NewLanguage 创建 Language API 分组。
func NewLanguage(s *session.Session) *Language {
	return &Language{ApiGroup: session.NewApiGroup(s)}
}

// SetCurrentLanguage 设置当前语言。对应 language/current-language (post_set)。
func (l *Language) SetCurrentLanguage(lang string) (interface{}, error) {
	return l.S.PostSet("language/current-language", map[string]interface{}{
		"CurrentLanguage": lang,
	}, false, "api", false, false)
}

// CurrentLanguage 获取当前语言。对应 language/current-language。
func (l *Language) CurrentLanguage() (map[string]interface{}, error) {
	return l.S.Get("language/current-language", nil, "api")
}

File: src/api/language.rs
//! Language API group (`api/Language.py`).

use serde_json::Value;

use crate::connection::Connection;
use crate::errors::Result;
use crate::tools::map_of;
use crate::xml::XmlMap;

use super::{get_value, post_set};

/// Language API group.
pub struct Language<'a> {
    conn: &'a Connection,
}

impl<'a> Language<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        Language { conn }
    }

    /// Set the current language.
    ///
    /// * `current_language` — language code (e.g. `"en_us"`).
    pub fn set_current_language(&self, current_language: &str) -> Result<String> {
        let body: XmlMap = map_of([("CurrentLanguage", current_language.to_string())]);
        post_set(self.conn, "api/language/current-language", &body)
    }

    /// `language/current-language`. Current language.
    pub fn current_language(&self) -> Result<Value> {
        get_value(self.conn, "api/language/current-language")
    }
}
